src/main/java/mate/academy/spring/security/CustomUserDetailsServiceImpl.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.spring.security;
+
+import static org.springframework.security.core.userdetails.User.withUsername;
+
+import java.util.Optional;
+import mate.academy.spring.model.Role;
+import mate.academy.spring.model.User;
+import mate.academy.spring.service.UserService;
+import org.springframework.security.core.userdetails.User.UserBuilder;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+public class CustomUserDetailsServiceImpl implements UserDetailsService {
+    private final UserService userService;
+
+    public CustomUserDetailsServiceImpl(UserService userService) {
+        this.userService = userService;
+    }
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        Optional<User> optionalUser = userService.findByEmail(username);
+        if (optionalUser.isEmpty()) {
+            throw new UsernameNotFoundException("User not found");
+        }
+        User user = optionalUser.orElseThrow();
```

Comment:
simplify this code using  `.orElseThrow(() ->
                new UsernameNotFoundException("Can't find user with " + "email: " + email));`
you don't need additional check isEmpty() it is duplication.